src/metatrain/experimental/alchemical_model/trainer.py

```diff
@@ -43,12 +43,14 @@ def __init__(self, train_hypers):
     def train(
         self,
         model: AlchemicalModel,
+        dtype: torch.dtype,
         devices: List[torch.device],
         train_datasets: List[Union[Dataset, torch.utils.data.Subset]],
         val_datasets: List[Union[Dataset, torch.utils.data.Subset]],
         checkpoint_dir: str,
     ):
-        dtype = train_datasets[0][0]["system"].positions.dtype
+        assert dtype in AlchemicalModel.__supported_dtypes__
```

Comment:
I think this can go outside of the trainers, in our infrastructure code rather than in the model code (same for the other architectures)

Reply:
We already do this outside. I just added this is in case one wants to debug the model without going through the whole metatrain infrastructure.

Reply:
Ok, fair